x-pack/plugins/ml/server/models/job_validation/job_validation.ts

```diff
@@ -100,6 +100,14 @@ export async function validateJob(
           ...(await validateModelMemoryLimit(client, mlClient, job, duration))
         );
       }
+
+      // if datafeed has aggregation, require job config to include a valid summary_doc_field_name
+      const datafeedAggConfig = job.datafeed_config?.aggregations ?? job.datafeed_config?.aggs;
```

Comment:
Is it possible to catch this validation on the page where you configure the detector? It works great on the validation step, but just wondering :)

![image](https://user-images.githubusercontent.com/7405507/98258875-4a1f4800-1f79-11eb-802d-ce579cd89f92.png)

Reply:
Added validation message here https://github.com/elastic/kibana/pull/81923/commits/d96a119950e710b043eb3c89568be037cc203ed2

<img width="1009" alt="Screen Shot 2020-11-05 at 12 05 55" src="https://user-images.githubusercontent.com/43350163/98281698-b04ca080-1f62-11eb-96cc-a2c91e78b928.png">

Reply:
I think it's worth editing the help text of the Summary count field as part of this PR, as I don't think the current `e.g. {docCountParam}` adds any clarity. Suggest alternatives for these two bits of text:

![image](https://user-images.githubusercontent.com/7405507/98365965-20265e00-202b-11eb-8e4b-4f467b78f4c8.png)

@lcawl or @szabosteve might be able to suggest better alternatives!

Reply:
The shortest description I can think of for the "summary count field" is this:

> If the input data is aggregated, specify the field that contains the document count.

If you also want to add a link to the docs, I recommend this one, since it provides a list of requirements and limitations: https://www.elastic.co/guide/en/machine-learning/master/ml-configuring-aggregation.html

Reply:
@lcawl @peteharverson I have [updated the texts](https://github.com/elastic/kibana/pull/81923/commits/22f938c9fd601a9c413d7f303ad0975997f649e7) for this area to be as below:
<img width="845" alt="Screen Shot 2020-11-11 at 10 08 30" src="https://user-images.githubusercontent.com/43350163/98836061-deb8f880-2406-11eb-9e80-5370a035ce89.png">

With `aggregated` linking to https://www.elastic.co/guide/en/machine-learning/master/ml-configuring-aggregation.html

Reply:
The help text and link for the summary count field LGTM